### DISCUSSION TITLE
Add link field with multi source option to core

### DISCUSSION THREAD

**Discussion by alexanderloewe** [craftcms/cms#9916] [Ideas] [answered]
Adding a new field with the option to link to every kind of content should be a core function of craft CMS.

Because it is an indispensable option - not only in terms of creating links to other content, but also when it comes to inserting content from different content types via a field in specific places in the templates.

**Comment by davidhellmann** [comment]
@brandonkelly what are your thoughts here. I know there is LinkIt and TypedLinkField but we miss this as a core field a lot. It's required in each of our projects and it's a bit sad to not have this. Maybe @sebastian-lenz can jump into it here and it's a possibility to move TypedLinkField over to P&T as a core plugin like Redactor. Just my thoughts in an easy world :D 

For me this is a more important feature than Matrix in Matrix ¯\_(ツ)_/¯

**Reply by brandonkelly** [reply]
I don’t think we would add it in core until it’s possible to query elements across multiple element types simultaneously.

**Comment by sebastian-lenz** [comment]
As far as I know P&T is aware of the demand. I've already signaled that it is okay for me should there by any kind of port or transition to a core field.

**Comment by piotrpog** [comment]
This is something i almost always use when building some kind of matrix based content builder. Choosing either entry link or external url link, selecting alternative link text, and toggling target blank.
I am aware there is Typed Link Field plugin, but for something so crucial i would really prefer first party functionality.
Is it possible that this will be included with craft 5?
@brandonkelly

**Reply by alexanderbuergin** [reply]
I would love to see that.

**Reply by brandonkelly** [reply]
Agree that Craft should have something like this, but can’t promise we’ll have time to get to it. In the meantime, check out [Hyper](https://plugins.craftcms.com/hyper?craft4) by Verbb as well.

**Comment by cherrykoda** [comment]
Attempting to guide/condense this request to the older post: https://github.com/craftcms/cms/discussions/9062 as there's a lot of asks spread around.

**Reply by brandonkelly** [reply]
Agree!

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.3.0-beta.1 is out with a new Link field type 🎉 (#15251).

We’re targeting August 6 for the GA release. If you’d like to test ahead of time, change your `craftcms/cms` requirement in composer.json to:

```json
"craftcms/cms": "^5.3.0-beta.1",
```

and then run `composer update`.
